src/base/bittorrent/torrentinfo.cpp

```diff
@@ -305,10 +293,22 @@ int BitTorrent::TorrentInfo::fileIndex(const QString& fileName) const
     return -1;
 }
 
-void TorrentInfo::remapFiles(libtorrent::file_storage const &fileStorage)
+void TorrentInfo::stripRootFolder()
 {
-    if (!isValid()) return;
-    m_nativeInfo->remap_files(fileStorage);
+    if (filesCount() <= 1) return;
+
+    libtorrent::file_storage files = m_nativeInfo->files();
+
+    // Solution for case of renamed root folder
+    std::string testName = filePath(0).split('/').value(0).toStdString();
+    if (files.name() != testName) {
+        files.set_name(testName);
+        for (int i = 0; i < files.num_files(); ++i)
+            files.rename_file(i, files.file_path(i));
```

Comment:
@sledgehammer999, OK, I remembered this. In this case, the files are detached only from the storage name, so they are affected by move_storage, but not affected by the storage renaming. That is why this code is here. Otherwise we will not be able to delete the root folder by renaming the storage to an empty string.
@thalieht, thank you for testing.